Security update and basic template

stephaniehobson opened the pull request:
## Description
- Create a basic two column template.
- Update the /security section to use it.
- Convert security.less to .scss component files.
  - Evaluate classes and remove unused ones.
- I did not update `hall-of-fame.less` because it gets bundled with `mozilla-accordion.less` and I didn't think I could bundle a mix of pre-processed files. I can convert  `mozilla-accordion.less` to .scss if we want though.

## Issue / Bugzilla link

[Bug 1438580](https://bugzilla.mozilla.org/show_bug.cgi?id=1438580)

## Testing

- The intention is to re-use this in the policies and legal sections as well. Does this look reusable? (I also have it running on policies locally but wanted to keep this commit small).
- Are the naming conventions I used in basic.scss okay?
- Do the classes I added to the global pebbles CSS belong there?
- What needs more commenting? Where did I stop commenting mid-sentence? 😝 
- Needs a bit of design review.

stephaniehobson commented:
@alexgibson I tried to create the boiler plate page suggested in [task 1651](https://tree.taiga.io/project/malexis-mozillaorg-durable-team/task/1651). Is this along the lines of what you were thinking?

I applied it to the participation guidelines locally and it serves both sections well. I think it can also be used in /legal. Let me know if I've gone about it badly. This feels like a big addition to be making to a code base I don't know well yet.

stephaniehobson commented:
@craigcook I tagged you to look at the additions to the global pebbles files, it's okay for you to pass on the html changes.

stephaniehobson commented:
@vincejoyiv 

This is mostly about updating the wrapper on some old content, focus on the wrapper and not the text.

Design questions I had:
- How can I visually separate the page from the nav menu. I borrowed a gradient bar from internet health but I sort of hate it.
- How are the typography and borders in the right column? 
- The first paragraph of the [landing page](https://bedrock-demo-shobson.us-west.moz.works/en-US/security/) had an `intro` class on it, so I borrowed the first paragraph styles from the blog. Too much?

Here is what I'm trying to update:
https://www.mozilla.org/en-US/security/

You can see what I've done on the demo server:
https://bedrock-demo-shobson.us-west.moz.works/en-US/security/

vincejoyiv commented:
@stephaniehobson I'd 86 that gradient bar. The title text should be enough to separate the nav from the page. I'd recommend lining up the top of the right column with the top of the paragraph text, instead of the header. https://screenshots.firefox.com/DgttjZyy0SEWiKnQ/bedrock-demo-shobson.us-west.moz.works

Recommend 86ing the `intro` class too. I'm looking forward to killing that on the blog in the future. 

The typography is fine (not as good as it'll be once we _design system_ everything, but it's as good as it was on sandstone).

alexgibson commented:
Not related to this change (so fixing is totally optional), but the table on `/security/client-bug-bounty/` does not play well on small screens, with most of the content being hidden out of the viewport.

![screenshot-2018-3-15 mozilla security bug bounty program](https://user-images.githubusercontent.com/400117/37469392-111a2f9a-285d-11e8-9105-9e5eaac9b96d.png)

alexgibson commented:
May not be fixable in this PR, but some of the advisories suffer from missing closing tags, resulting in broken styling:

![screenshot-2018-3-15 arbitrary code execution through unsanitized browser ui](https://user-images.githubusercontent.com/400117/37469627-987e4cf0-285d-11e8-97de-821e1496e4aa.png)

alexgibson commented on media/css/pebbles/base/elements/_lists.scss:
```diff
@@ -59,3 +63,13 @@ ol.prose {
+    }
+    dd {
+        margin-bottom: .25em;
+        padding-left: 60px;
```
Visually, is this maybe a little much?

<img width="866" alt="screen shot 2018-03-15 at 14 37 48" src="https://user-images.githubusercontent.com/400117/37470009-79cdec6a-285e-11e8-9b1d-1348deae5962.png">

stephaniehobson replied:
Decreased to `$gutter-width`

alexgibson commented:
> I did not update hall-of-fame.less because it gets bundled with mozilla-accordion.less and I didn't think I could bundle a mix of pre-processed files. I can convert mozilla-accordion.less to .scss if we want though.

Looking at [hall-of-fame.less](https://github.com/mozilla/bedrock/blob/master/media/css/security/hall-of-fame.less), it's only a really small bit of CSS and does not use any Sandstone specific features. I think it may be fine to convert this file to .scss, evenif you don't want to touch `mozilla-accordion.less`?

stephaniehobson commented on bedrock/base/templates/base-pebbles-basic.html:
```diff
@@ -0,0 +1,26 @@
+{% extends 'base-pebbles.html' %}
+
+{% block site_css %}
+  {% stylesheet 'pebbles-basic' %}
```
✅

stephaniehobson commented on bedrock/base/templates/base-pebbles-basic.html:
```diff
@@ -0,0 +1,26 @@
+  </main>
+{% endblock %}
+
+{% block email_form %}{% endblock %}
```
✅

stephaniehobson commented on bedrock/base/templates/macros.html:
```diff
@@ -2,6 +2,41 @@
+
+{% macro side_nav_lists(arr, body_id='') -%}
+  {# TODO: it would be nice if this could take a single list of links #}
+  {# TODO: current page is not being highlighted #}
```
✅

stephaniehobson commented on bedrock/base/templates/macros.html:
```diff
@@ -2,6 +2,41 @@
+#}
+
+{% macro side_nav_lists(arr, body_id='') -%}
+  {# TODO: it would be nice if this could take a single list of links #}
```
👍

stephaniehobson commented on media/css/pebbles/components/_basic.scss:
```diff
@@ -0,0 +1,79 @@
+        padding: $gutter-width $desktop-gutter $desktop-gutter ($desktop-gutter / 2);
+    }
+
+    // RTL
```
✅

stephaniehobson commented on media/css/pebbles/components/_footer.scss:
```diff
@@ -7,6 +7,7 @@
 #colophon {
     @include open-sans;
     background: #000;
+    clear: both;
```
Not for this PR.

I added it because the first time I opened my new template the entire page was black 😖 

The page behaves properly without it now but this seems like a good win for avoiding unforeseen issues on pages with coding problems.

![screen shot 2018-03-15 at 14 43 08-fullpage](https://user-images.githubusercontent.com/854701/37492754-bfa71aee-285f-11e8-9a43-113283928532.png)

stephaniehobson commented on media/css/pebbles/includes/mixins/_typography.scss:
```diff
@@ -138,3 +138,11 @@
+@mixin intro-text {
+    @include font-size-level4;
+
+    &:first-line {
```
Removed as per Vince.

stephaniehobson commented on media/css/security/components/advisory.scss:
```diff
@@ -0,0 +1,45 @@
+
+.advisory {
+    .summary {
+        @include clearfix;
```
No, now that this is a grid it works fine without it 😄 Grid FTW.

stephaniehobson commented on media/css/pebbles/components/_basic.scss:
```diff
@@ -0,0 +1,79 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@import '../includes/lib';
+
```
Thanks for explaining this. I can't wait for Protocol to remove some of the confusion.

stephaniehobson commented on media/css/pebbles/components/_basic.scss:
```diff
@@ -0,0 +1,79 @@
+/* one large column on the left, one optional smaller column on the right */
+
+$mobile-gutter: $gutter-width;
+$tablet-gutter: 60px;
```
✅ Re-written

stephaniehobson commented:
This is ready for re-review. With the work on the manifesto page we need to update https://www.mozilla.org/en-US/about/manifesto/details/. I'd love to be able to use this template for that, which means the urgency of merging this has increased (EOD Monday is fine).

The unclosed tag that was messing with the advisory formatting seems to be fixed now. I wasn't able to reproduce it locally.

I converted hall-of-fame.less to .scss.

And incorporated the design feedback.

I'll have a look at that awful table tomorrow if I have time, it can be dealt with in a separate PR if necessary.

alexgibson commented:
Now that the horizontal rule has been removed, the main heading could maybe use a little more top/bottom padding on large viewports. Here's a rough comparison to the `/contact/` page:

![header-padding](https://user-images.githubusercontent.com/400117/37512936-5e780188-28fb-11e8-9b98-d8688afee708.png)

alexgibson commented:
For non-English locales which don't get the global nav, Tabzilla is still visible at mobile viewport sizes:

![tabzilla](https://user-images.githubusercontent.com/400117/37513154-1c921154-28fc-11e8-844e-39daff03ed3f.png)

I'd suggest adding `{% block tabzilla_tab %}{% endblock %}` to the pebbles-basic base template to override this. So. Much. Legacy. Debt. Bleurgggh 😩

alexgibson commented:
To get a version of the legacy nav that has more up-to-date styles for non-English locales, 

Add this to `base-pebbles-basic.html`:

```
{% block site_header %}
  {% if LANG.startswith('en-') %}
    {% include 'includes/global-nav.html' %}
  {% else %}
    {% set logo_src = static('img/pebbles/moz-wordmark-dark-reverse.svg') %}
    {% include 'mozorg/home/includes/nav.html' %}
  {% endif %}
{% endblock %}
```

And this to `basic.scss`:

```css
@import '../hubs/masthead';
```

alexgibson commented:
The main content has some extra left padding that can be removed. The main title should sit flush with the Mozilla logo in the header, and the right edge of the sidebar aligned with the download button.

![padding](https://user-images.githubusercontent.com/400117/37513963-6a2f258a-28fe-11e8-93e5-0c27cad4182d.png)

Please excuse my messy desktop 😄

stephaniehobson commented:
Updated.

alexgibson commented:
@stephaniehobson looks like this needs a rebase after another PR that just merged.

stephaniehobson commented:
Rebased.
